File: src/contest/winter_2022/jan_15/a/Main.cpp
// https://www.51nod.com/Challenge/Problem.html#problemId=3056
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n, k;
  cin >> n >> k;
  vector<int> ls;
  for (int i = 0; i < n; i++) {
    int tmp;
    cin >> tmp;
    ls.push_back(tmp);
  }
  ls.resize(n);
  for (int i = 0; i < k; i++) {
    char o;
    cin >> o;
    int x, y;
    switch (o) {
      case 'Z':
        cin >> x >> y;
        if (x > ls.size()) {
          ls.push_back(y);
        } else {
          ls.insert(ls.begin() + x, y);
        }
        break;
      case 'D':
        cin >> x;
        if (x >= ls.size())
          continue;
        ls.erase(ls.begin() + x);
        break;
    }
  }
  cout << ls.size() << endl;
  for (int i = 0; i < ls.size(); i++) {
    cout << ls[i] << " ";
  }
  return 0;
}

File: src/contest/winter_2022/jan_15/a/Main.py
# https://www.51nod.com/Challenge/Problem.html#problemId=3056
(n, k) = list(map(int, input().split()))
ls = list(map(int, input().split()))
for i in range(k):
    s = input().split()
    if s[0] == 'Z':
        x = int(s[1])
        y = int(s[2])
        if x > len(ls):
            ls.append(y)
        else:
            ls.insert(x, y)
    elif s[0] == 'D':
        x = int(s[1])
        if (x >= len(ls)):
            continue
        del ls[x]
print(len(ls))
for i in ls:
    print(i, end=' ')
